### PR TITLE
Update Quarkus to 3.14.2

### PR DESCRIPTION
**Pull request by fedinskiy**
- Update Quarkus dependencies from 3.6 ro 3.14
- Use new names of reactive extensions
- Update Quinoa to the latest verison (the old one failed with Quarkus 3.14)

Fixes https://github.com/quarkusio/quarkus-workshops/issues/587

### CODE REVIEW

**Review by geoand — APPROVED** [review]
Thanks!

**Comment by fedinskiy** [comment]
`   Aborted - you do not have permission to publish to quarkiverse-quarkus-workshops-pr-588-preview.surge.sh`  [1]

This is probably unrelated, but I also add change, related to kafka

[1] https://github.com/quarkusio/quarkus-workshops/actions/runs/10738710976/job/29782851900

File: quarkus-workshop-super-heroes/docs/.jbang/jbang/bin/jbang [hunk 1]
```diff
@@ -0,0 +1,230 @@
+#!/usr/bin/env bash
```
Comment (geoand) [inline quarkus-workshop-super-heroes/docs/.jbang/jbang/bin/jbang:1]: What's the deal with this jbang stuff?

Comment (fedinskiy) [reply quarkus-workshop-super-heroes/docs/.jbang/jbang/bin/jbang]: It was a mistake, removed it

**Review by geoand — APPROVED** [review]
Thanks!

**Comment by fedinskiy** [comment]
@geoand the CI is green, could you merge it?

**Pull request merged**